src/web/src/hooks/use-telescope-contributor.ts

```diff
@@ -12,31 +11,80 @@ export type Contributor = {
 const telescopeGitHubContributorsUrl =
   'https://api.github.com/repos/Seneca-CDOT/telescope/contributors';
 
-const randomContributor = (contributors?: any[]) =>
-  contributors ? contributors[Math.floor(Math.random() * contributors.length)] : null;
+const prepareURL = (page?: number) =>
+  `${telescopeGitHubContributorsUrl}?per_page=1&${page ? `page=${page}` : ''}`;
 
-const PER_PAGE: number = 50;
+const linkParser = (links: string, key: string) => {
+  let url;
+  let page;
+  const regex = new RegExp(`<([^?]+\\?per_page=1&[a-z]+=([\\d]+))>;[\\s]*rel="${key}"`, 'g');
+  const arr = regex.exec(links);
 
-const useTelescopeContributor = () => {
-  const [contributor, setContributor] = useState<Contributor | null>(null);
+  if (arr !== null) {
+    [, url, page] = arr;
+  }
+
+  return { url, page };
+};
+
+const getMaxContributors = async () => {
+  const response = await fetch(prepareURL());
+  const links = response.headers.get('Link') || '';
+
+  // Extract page count from response headers
+  const { page: max_contributors } = linkParser(links, 'last');
+  if (window.localStorage) window.localStorage.setItem('fetch_date', String(Date.now()));
```

Comment:
Prefer `useLocalStorage` hook like I did in https://github.com/Seneca-CDOT/telescope/pull/1843/files#diff-b561b0683cdcdae7324e2618b3f251e88a961a8160768296a03ab3fb9723f895.  NOTE: if you rely on `window`, we'll have to sort out server side rendering too, where it doesn't exist.